src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,45 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 1;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length > THRESHOLD) {
+            int middle = array.length / 2;
+            int[] left = Arrays.copyOfRange(array, 0, middle);
+            int[] right = Arrays.copyOfRange(array, middle, array.length);
+            MergeSortAction leftAction = new MergeSortAction(left);
+            MergeSortAction rightAction = new MergeSortAction(right);
+            invokeAll(leftAction, rightAction);
+            merge(left, right);
```

Comment:
The merge method is being called without actually merging the sorted halves back into the original 'array'. After the two RecursiveActions for the left and right halves have been invoked, the results should be merged back into the original array to complete the sorting process.